Java/parkservices/src/main/java/com/amazon/randomcutforest/extendedrandomcutforest/threshold/ThresholdedRandomCutForest.java

```diff
@@ -0,0 +1,406 @@
+            lastAnomalyTimeStamp = result.timeStamp;
+            lastAnomalyPoint = Arrays.copyOf(point,point.length);
+            update(result.rcfScore, result.rcfScore, true);
+        } else {
+            /**
+             * we again check if the new input produces an anomaly/not on its own
+             */
+            if (trigger(result.attribution, gap, baseDimensions, newAttribution) && result.rcfScore > newScore) {
+                result.anomalyGrade = thresholder.getAnomalyGrade(result.rcfScore);
+                lastAnomalyScore = result.rcfScore;
+                lastAnomalyAttribution = new DiVector(result.attribution);
+                lastAnomalyTimeStamp = result.timeStamp;
+                lastAnomalyPoint = Arrays.copyOf(point,point.length);
+                update(result.rcfScore, result.rcfScore, true);
+            } else {
+                // not changing inAnomaly
+                result.anomalyGrade = 0;
+                update(result.rcfScore, result.rcfScore,false);
+            }
+        }
+
+        if (result.anomalyGrade>0){
+            result.expectedValuesPresent = reasonableForecast;
+            if (result.relativeIndex < 0 && result.startOfAnomaly) {
+                // anomaly in the past and detected late; repositioning the computation
+                startPosition = result.attribution.getDimensions() + (result.relativeIndex - 1) * baseDimensions;
+                if (result.expectedValuesPresent) {
+                    int[] missingIndices = largestFeatures(result.attribution, startPosition, baseDimensions, numberOfAttributors);
+                    newPoint = forest.imputeMissingValues(point, missingIndices.length, missingIndices);
+                    result.oldValues = new double[baseDimensions];
+                    for (int i = 0; i < baseDimensions; i++) {
+                        result.oldValues[i] = point[startPosition + i];
+                    }
+                }
+            }
+            if (result.expectedValuesPresent) {
+                result.expectedValuesList = new double[1][];
+                result.expectedValuesList[0] = new double[baseDimensions];
+                for (int i = 0; i < baseDimensions; i++) {
+                    result.expectedValuesList[0][i] = newPoint[startPosition + i];
+                }
+                result.likelihoodOfValues = new double[]{1.0};
+                lastExpectedPoint = Arrays.copyOf(newPoint,newPoint.length);
+            } else {
+                lastExpectedPoint = null;
+            }
+
+            result.flattenedAttribution = new double[baseDimensions];
+            for (int i = 0; i < baseDimensions; i++) {
+                result.flattenedAttribution[i] = result.attribution.getHighLowSum(startPosition + i);
+            }
+        }
+        return result;
+    }
+
+    public boolean isPreviousIsPotentialAnomaly() {
+        return previousIsPotentialAnomaly;
+    }
```

Comment:
You could replace all these simple getters with lombok annotations if you like.

Reply:
This is a fantastic suggestion!